client/settings/settings.tsx

```diff
@@ -1,228 +1,370 @@
-    setTempLocalSettings(prev => prev.merge(settings))
-    setTempScrSettings(prev => prev.merge(settings))
-  }, [])
-
-  const onSettingsSubmit = useCallback(() => {
-    // NOTE(tec27): We remove the StarCraft path here because we don't provide a way to change it
-    // in this dialog anyway, and depending on how IPC events interleave, the SC Path dialog may
-    // not have merged a new path by the time this dialog is first rendered. We don't want to merge
-    // an old path back into the settings, so we just remove it instead.
-    const localSettingsToMerge = tempLocalSettings.toJS() as Partial<LocalSettingsData>
-    delete localSettingsToMerge.starcraftPath
-    dispatch(mergeLocalSettings(localSettingsToMerge))
-    dispatch(mergeScrSettings(tempScrSettings.toJS() as Partial<ScrSettingsData>))
-
-    // TODO(tec27): This doesn't seem like it would actually catch errors from saving here? Since
-    // those would happen async-ly. Should probably have a form of this that doesn't bother with
-    // dispatching and just returns a promise we can await
-    if (!lastError) {
-      dispatch(closeDialog(DialogType.Settings))
-    }
-  }, [dispatch, tempLocalSettings, tempScrSettings, lastError])
-
-  useEffect(() => {
-    saveButtonRef.current?.focus()
-  }, [])
-
-  const starcraftVersionText = 'StarCraft: Remastered'
-  const titleAction = (
-    <TitleActionContainer>
-      <TitleActionText onClick={onSetPathClick}>{starcraftVersionText}</TitleActionText>
-      <TitleActionButton
-        icon={<SetPathIcon />}
-        title='Change StarCraft path'
-        onClick={onSetPathClick}
-      />
-    </TitleActionContainer>
-  )
+      return false
+    },
+  })
 
-  const tabs = (
-    <Tabs activeTab={activeTab} onChange={onTabChange}>
-      <TabItem key='app' text='App' value={SettingsTab.App} />
-      <TabItem key='input' text='Input' value={SettingsTab.Input} />
-      <TabItem key='sound' text='Sound' value={SettingsTab.Sound} />
-      <TabItem key='video' text='Video' value={SettingsTab.Video} />
-      <TabItem key='gameplay' text='Gameplay' value={SettingsTab.Gameplay} />
-    </Tabs>
-  )
+  // TODO(2Pac): Currently we don't have any non-Electron settings so this might look a bit weird.
+  // Soon enough we should probably have some user settings (e.g. editing user account, and changing
+  // a language).
 
-  const buttons = [
-    <TextButton label='Cancel' key='cancel' color='accent' onClick={onSettingsCancel} />,
-    <TextButton
-      ref={saveButtonRef}
-      label='Save'
-      key='save'
-      color='accent'
-      onClick={onSettingsSave}
-    />,
-  ]
-
-  let contents: React.ReactNode
-  switch (activeTab) {
-    case SettingsTab.App:
-      contents = (
-        <AppSettings
-          localSettings={tempLocalSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
+  let contents: React.ReactNode | undefined
```

Comment:
This pattern is imo a mess, write a component that has both of these values on it and does the right thing. Bonus points for making it something that returns from the switch instead of mutating a variable that is used later.